File: t/TypeScript/src/commands/KickCommand.ts
import { Message } from "discord.js";
import { Command } from "../lib/commands/Command";

export class KickCommand extends Command {
	execute(
		message: Message,
		commandName: string,
		...args: string[]
	): Promise<any> {
		const user = message.mentions.users.first();

		if (!user) {
			message.reply("You did not mention a user to kick!");
		}

		const member = message.guild.member(user);

		if (!member) {
			message.reply("That user is not in this guild!");
		}

		const reason = args.slice(1).join(" ");
		member
			.kick(args.join(" "))
			.then(() => {
				message.reply(`Kicked ${user} for reason: ${reason}`);
			})
			.catch(() => {
				message.reply(
					`I was unable to kick ${user}, I may not have the required permissions to do this!`
				);
			});
		return;
	}
}

File: k/kotlin/src/main/kotlin/sh/foxboy/commands/KickCommand.kt
package sh.foxboy.commands

import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.entities.Member
import net.dv8tion.jda.api.entities.Message
import sh.foxboy.api.command.Command


class KickCommand(name: String) : Command(name) {
    override fun execute(message: Message, commandLabel: String, args: Array<String?>?) {
        if (message.mentionedMembers.isEmpty()) {
            message.reply("You must mention a user to kick!").queue()
            return
        }

        val sender: Member? = message.guild.getMember(message.author)
        val target: Member = message.mentionedMembers[0]

        if (sender?.hasPermission(Permission.KICK_MEMBERS) != true || !sender.canInteract(target)
            || !message.guild.selfMember.canInteract(target)
            || !message.guild.selfMember.hasPermission(Permission.KICK_MEMBERS)
        ) {
            message.replyFormat(
                "I was unable to kick %s, I may not have the required permissions to do this!",
                target.user.asMention
            ).queue()
            return
        }

        target.kick(args?.joinToString()).queue(
            {
                message.replyFormat(
                    "Kicked %s for reason: %s", target.user.asMention,
                    args?.joinToString()
                )
            }
        ) { message.replyFormat("An error occurred while trying to kick %s", target.user.asMention) }
    }
}